TikTok Ads integration doc (#364)
rudderlabs/rudderstack-docs: destinations/advertising/tiktok-ads.mdx

---
# slug: "/docs/destinations/advertising/tiktok-ads"
title: "TikTok Ads"
description: Step-by-step guide to send your event data from RudderStack to TikTok Ads.
---

# TikTok Ads

> <a href="https://www.tiktok.com/business/ru/apps/tiktok?tt4b_lang_redirect=1">TikTok Ads</a> is TikTok's online advertising platform. It allows marketers to run their product ads to target custom audiences, optimize their campaigns, attribution, etc.

Reviewer: ```suggestion
<a href="https://www.tiktok.com/business/en/apps/tiktok?tt4b_lang_redirect=1">TikTok Ads</a> is TikTok's online advertising platform. It allows marketers to run their product ads to target custom audiences, optimize their campaigns, attribution, etc.
```
The provided URL is in the Russian language

RudderStack supports TikTok Ads as a destination where you can send your event data seamlessly.

<div class="successBlock">

  <strong>Find the open-source transformer code for this destination in our <a href="https://github.com/rudderlabs/rudder-transformer/tree/master/v0/destinations/tiktok_ads">GitHub repo</a></strong>.
</div>

## Getting started

Before configuring TikTok Ads as a destination in RudderStack, verify if the source platform is supported by TikTok Ads by referring to the table below:

| **Connection Mode** | **Web**       | **Mobile**    | **Server**    |
| :------------------ | :------------ | :------------ | :------------ |
| **Device mode**     |  -            | -             | -             |

> | **Cloud mode**      | **Supported** | -             | -             |

Reviewer 2: @akashrpo Why are "Mobile" and "Server" left out here?

Reviewer: @ishachopra0401 Cloud mode: `Mobile` and `Server` will be supported fields here.

- **Pixel Code**: You can find the Pixel Code by referring to <a href="https://ads.tiktok.com/gateway/docs/index?identify_key=2b9b4278e47b275f36e7c39a4af4ba067d088e031d5f5fe45d381559ac89ba48&language=ENGLISH&doc_id=1701890979375106#item-link-Where%20to%20Find%20pixel_code">TikTok documentation</a>.

- **Hash Contextual User Properties (SHA-256)**: If this setting is enabled, RudderStack will hash the contextual user properties like external_id, email, phone_number from raw format to sha256.

### Track

The `track` call allows you to capture user actions along with the properties associated with these actions. Each action is considered to be an event.

A sample `track` call is shown below:

```javascript
rudderanalytics.track(
  "checkout started", {
    eventId: "1616318632825_357",
    context: {
      ad: {
        callback: "123ATXSfe"
      }
    },
    currency: "USD",
    value: 46.0
  }, {
    externalId: [{
      type: "tiktokExternalId",

>       id: "f0e388f53921a51f0bb0fc8a2944109ec188b59172935d8f23020b1614cc44bc"

Reviewer: Let's update this field to
```suggestion
      id: "1234"
```
as by default, we will hash this property

    }]
  }
);
```
<div class="infoBlock">

> 

Reviewer: We need to add the mapping table below for events that we are mapping.

> The events are automatically grouped in a batch of 50 events per request using the <code class="inline-code">batch</code> call. However, if the payload contains <code class="inline-code">test_event_code</code> property, batching won't be applied. For more information on the `batch` call, please refer to the <a href="https://www.rudderstack.com/docs/rudderstack-api/http-api/#11-batch">RudderStack API Specification</a> documentation.

Reviewer 3: @akashrpo @ishachopra0401 is not able to make many contexts out of this paragraph, as the prior example is given using JS SDK, and in case batching JS SDK doesn't support batching. Probably we can use any server-side SDK example for this doc, and give example for batching as well?
cc @saikumarrs  
WDYT?

Reviewer 2: I don't think this is applicable here. Looking for more context and the need.

Reviewer 2: SDK batching functionality is usually discussed in the respective SDK's documentation.

Reviewer: IMHO along with the provided JS-SDK example, we can also provide another example for batching that will give more context to the user. Perhaps we can also modify this sentence to say:

Rudderstack will automatically group all the events which are received in a batch of 50 per request and send using the `batch` endpoint. However, if the payload contains `test_event_code` property, batching won't be applied. For more information on the `batch` call, please refer to the <a href="https://www.rudderstack.com/docs/rudderstack-api/http-api/#11-batch">RudderStack API Specification</a> documentation. 
cc @ishachopra0401

Reviewer 3: @ishachopra0401 let's add link to our batch API spect to add some more context https://www.rudderstack.com/docs/rudderstack-api/http-api/#11-batch

Author: done, already!

Reviewer 2: @ameypv-rudder @ishachopra0401 this paragraph is not needed in destination documentation as it is irrelevant here.
@akashrpo to provide clarification on this.

</div>

> 

Reviewer: Let's add the FAQ section below.

Reviewer 4: @akashrpo since the FAQ (re the 40002 error) is not specifically related to the RudderStack integration with TikTok, do we need to add it?

Reviewer: As discussed I think the user's first approach will be to look for the solution to the error at our end. In which case we can add this piece of information in our docs.
cc @utsabc

---
# slug: "/docs/destinations/advertising/tiktok-ads"
title: "TikTok Ads"
description: Step-by-step guide to send your event data from RudderStack to TikTok Ads.
---

# TikTok Ads

> [TikTok Ads](https://www.tiktok.com/business/en/apps/tiktok?tt4b_lang_redirect=1") is TikTok's online advertising platform. It allows marketers to run their product ads to target custom audiences, optimize their campaigns, attribution, etc.

Reviewer 4: ```suggestion
[TikTok Ads](https://www.tiktok.com/business/en/apps/tiktok?tt4b_lang_redirect=1") is TikTok's online advertising platform. It lets marketers run their product ads to target custom audiences and optimize their campaigns.
```

| **Device mode**     |  -            | -             | -             |
| **Cloud mode**      | **Supported** | **Supported** | **Supported** |

<div class="infoBlock">

To know more about the difference between cloud mode and device mode in RudderStack, refer to the <a href="https://rudderstack.com/docs/connections/rudderstack-connection-modes/">RudderStack Connection Modes</a> guide.
</div>

Once you have confirmed that the source platform supports sending events to TikTok Ads, follow these steps:

- From your [**RudderStack dashboard**](https://app.rudderstack.com/), add the source. Then, from the list of destinations, select **TikTok Ads**.

<div class="infoBlock">

Follow our guide on <a href="https://rudderstack.com/docs/connections/adding-source-and-destination-rudderstack/">Adding a Source and Destination in RudderStack</a> for more information.
</div>

<img src="../../assets/event-stream-destinations/tiktokads-connection-settings.png" alt="TikTok ads connection settings"/>

## Connection settings

To successfully configure TikTok Ads as a destination, you will need to configure the following settings:

> - **Access Token**: You need to generate the Access Token (also known as long-term access token) by following the steps mentioned in <a href="https://ads.tiktok.com/marketing_api/docs?id=1701890912382977">Authorization</a> and <a href="https://ads.tiktok.com/marketing_api/docs?id=1701890914536450">Authentication</a> topics in TikTok documentation.

Reviewer 4: ```suggestion
- **Access Token**: You need to generate the Access Token (also known as Long Term Access Token) by following the steps mentioned in <a href="https://ads.tiktok.com/marketing_api/docs?id=1701890912382977">Authorization</a> and <a href="https://ads.tiktok.com/marketing_api/docs?id=1701890914536450">Authentication</a> topics in TikTok documentation.
```

Reviewer 4: In the web app, we reference the access token as Long Term Access Token, and so do Segment. We can be consistent here.

Author: Going by the official tiktok ads doc here.

- **Pixel Code**: You can find the Pixel Code by referring to <a href="https://ads.tiktok.com/gateway/docs/index?identify_key=2b9b4278e47b275f36e7c39a4af4ba067d088e031d5f5fe45d381559ac89ba48&language=ENGLISH&doc_id=1701890979375106#item-link-Where%20to%20Find%20pixel_code">TikTok documentation</a>.

> - **Hash Contextual User Properties (SHA-256)**: If this setting is enabled, RudderStack will hash the contextual user properties like external_id, email, phone_number from raw format to sha256.

Reviewer 4: ```suggestion
- **Hash Contextual User Properties (SHA-256)**: If this setting is enabled, RudderStack will hash the contextual user properties like `external_id`, `email`, `phone_number` from the raw format to the SHA-256 format.
```
